src/main/java/com/fundynamic/d2tm/game/map/Map.java

```diff
@@ -123,4 +135,54 @@ public Cell getCell(Vector2D mapCoordinates) {
         return getCell(mapCoordinates.getXAsInt(), mapCoordinates.getYAsInt());
     }
 
+    public String getAsciiShroudMap() {
+        String result = "";
+        for (int y = 0; y < height; y++) {
+            String line = "";
+            for (int x = 0; x < width; x++) {
+                if (getCell(x, y).isShrouded()) {
+                    line += "#";
+                } else {
+                    line += ".";
+                }
+            }
+            result += line + "\n";
+        }
+        return result;
+    }
+
+    public void revealShroudFor(int x, int y) {
+        getCell(x, y).setShrouded(false);
+    }
+
+    public void revealShroudFor(int x, int y, int range) {
+        if (range < 1) return;
+
+
+        float halfATile = TILE_SIZE / 2;
+        // convert to absolute pixel coordinates
+        Vector2D asPixelsCentered = getCellCoordinatesInAbsolutePixels(x, y).
+                add(Vector2D.create(halfATile, halfATile));
+
+        double centerX = asPixelsCentered.getX(), centerY = asPixelsCentered.getY();
+
+        for (int rangeStep=0; rangeStep < range; rangeStep++) { // range 'steps'
+
+            for (int degrees=0; degrees < 360; degrees++) {
+
+                // calculate as if we would draw a circle and remember the coordinates
+                float rangeInPixels = (rangeStep * TILE_SIZE);
+                double radians = Math.toRadians(degrees);
+
+                double circleX = (centerX + (Math.cos(radians) * rangeInPixels));
+                double circleY = (centerY + (Math.sin(radians) * rangeInPixels));
```

Comment:
ah well since we probably have a finite amount of ranges (most of the time) we can probably store them as well. But I see these operations are cheap compared to the `cos` and `sin` functions :)